aspnetcore/fundamentals/host/platform-specific-configuration/samples/2.x/HostingStartupApp/HostingStartupApp.csproj

```diff
@@ -0,0 +1,36 @@
+<Project Sdk="Microsoft.NET.Sdk.Web">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp2.1</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.App" Version="2.1.3" />
+    <PackageReference Include="HostingStartupPackage" Version="1.0.0" />
+  </ItemGroup>
+
+  <!--
+    The following PropertyGroup sets the package restore sources for the app.
+
+    The HostingStartupPackage bin/Debug folder is added.
+    
+    This permits the app to restore the HostingStartupPackage from its 
+    project's published output. There is no need to publish the package to 
+    nuget.org.
+  -->
+  <PropertyGroup>
+    <RestoreSources>$(RestoreSources);https://api.nuget.org/v3/index.json;../HostingStartupPackage/bin/Debug</RestoreSources>
+  </PropertyGroup>
+
+  <!--
+    The following ItemGroup permits the app to load the HostingStartupLibrary
+    from the app's bin folder.
+  -->
+  <ItemGroup>
+    <Reference Include=".\bin\Debug\netcoreapp2.1\HostingStartupLibrary.dll">
```

Comment:
@dasMulli,
It does seem hacky to hardcode certain pieces of this path. Should they be converted to variables? For example:
- Debug --> $(Configuration)
- netcoreapp2.1 --> $(TargetFramework)
- HostingStartupLibrary.dll --> $(AssemblyName)

Reply:
I think this is a great idea. We do that elsewhere, and we probably should do it here, too.

I'll use different names. They refer to characteristics of another project (i.e., the hosting startup package project). I'll make it clear in their names on the next commit.

Reply:
why is the reference here in the first place? shouldn't it be coming from the `HostingStartupPackage` package reference?

Reply:
idk on this now ....... this is taking something very simple and making it *very messy*. It takes three vars for both to define them. It makes it harder to see what it is .... just a path. I'd rather be super clear on the components of the path than have a bunch of vars stuck in here.

Reply:
So.. one is a package reference and the other a `Reference` instead of a project-to-project reference..

I'd argue that the `Reference` should really be a `ProjectReference` instead, so it is built, up-to-date and you get the correct path.

Reply:
btw i'd prefer `<Reference Include="path/to/some.dll" />` instead of the hint path approach.

Reply:
Yes on the first one. The package is a different demo from the library. These are two different  compile-time approaches.

On the second point ... The topic's demos include the *deployment strategy*. It's similar to our earlier discussion on why I'm working with package sources and not a project reference to the hosting startup package project. Same thing here: AFAICT from what the engineers have delivered for this scenario, the hosting startup is **_deployed_** as a DLL. Therefore, this doesn't use a project ref. It references the DLL directly (and this will change for any given organization/environment) ... or the DLL will be dropped directly into the consuming project.

I think what's missing here is a little more guidance in the project file. Let me see what I can do on the next commit with the explanation. I'll check the topic, too, to see what I need to do there.

Reply:
Ah yeah so in theory it could also be a `<Content Include="../startup.dll" CopyToOutputDirectory="PreserveNewest" />`.
(however, MSBuild also considers `@(Content)` for assembly resolution which is kind of weird).

Reply:
Sorry about that @scottaddie ... I just noticed that you are correct about the instructions and location for the hosting startup library. The instructions are correct ... I didn't recall that I have the dev deploy the assembly to the consuming project. These scenarios aren't exactly easy to track. I'll add project file language to reinforce the concept, and I'll add the vars you recommended.

☕️ ....... *one more cup please!*

Reply:
>Debug --> $(Configuration)
>netcoreapp2.1 --> $(TargetFramework)
>HostingStartupLibrary.dll --> $(AssemblyName)

All of these only apply when the referenced project is built under the exact same parameters.
* `$(Configuration)` -> Probably okay if they are built in the same solution and users didn't mess with solution configurations.
* `$(TargetFramework)` -> as long as the library isn't a `netstandard*`...
* `$(AssemblyName)` -> only if the referenced project's `.csproj` file has the exact same name as the currently built project.. don't think that will happen

Reply:
> All of these only apply when the referenced project is built under the exact same parameters.

It's just for the path tho (within the consuming project; I mean the configuration and the target framework vars), so the path to the DLL should be correct.

I need to think this over a little more. Everything we've discussed has merit, but this is on a slippery slope to being so ✨ _magical_ ✨ that it clouds the basic concepts.

Reply:
there's also..
```
<ProjectReference Include="../some/other.csproj" ReferenceOutputAssembly="false" />
```

Reply:
I'm fleshing out some details in the app's project file to explain what's going on.

I may also mirror some or all of the new remarks in the topic itself.

It will be on the next commit.